src/logic/wallets/store/selectors/index.ts

```diff
@@ -2,28 +2,30 @@ import { createSelector } from 'reselect'
 
 import { ETHEREUM_NETWORK, ETHEREUM_NETWORK_IDS } from 'src/logic/wallets/getWeb3'
 import { PROVIDER_REDUCER_ID } from 'src/logic/wallets/store/reducer/provider'
+import { GnosisState } from 'src/store'
+import { ProviderRecord } from '../model/provider'
 
-export const providerSelector = (state) => state[PROVIDER_REDUCER_ID]
+export const providerSelector = (state: GnosisState): ProviderRecord => state[PROVIDER_REDUCER_ID]
 
-export const userAccountSelector = createSelector(providerSelector, (provider) => {
+export const userAccountSelector = createSelector(providerSelector, (provider: ProviderRecord): string => {
   const account = provider.get('account')
-
   return account || ''
 })
 
-export const providerNameSelector = createSelector(providerSelector, (provider) => {
+export const providerNameSelector = createSelector(providerSelector, (provider: ProviderRecord): string | undefined => {
   const name = provider.get('name')
-
   return name ? name.toLowerCase() : undefined
 })
 
-export const networkSelector = createSelector(providerSelector, (provider) => {
+export const networkSelector = createSelector(providerSelector, (provider: ProviderRecord): string => {
```

Comment:
can return type be changed to `ETHEREUM_NETWORK_IDS[number]` or something like that?

Reply:
that exact same suggestion didn't work, I think the way to fix this will be to redeclare those `const` objects as `enum`. Won't do that in this PR.